daemon/restart.go

```diff
@@ -18,7 +18,7 @@ func (daemon *Daemon) ContainerRestart(name string, seconds *int) error {
 	if err != nil {
 		return err
 	}
-	if seconds == nil {
+	if seconds == nil || *seconds < 0 {
```

Comment:
@miaoyq If check the validate in API router, we can not get the `default timeout of container` and `stopTimeout`

So I think this is better. and here check the `seconds ==nil`